### DISCUSSION TITLE
How do I add cursive ocr capabilities?

### DISCUSSION THREAD

**Discussion by kmgbarroga** [robertknight/ocrs#87] [Q&A] [answered]
Hi, sorry, the whole concept of training and adding models are very new to me. As far as I know, the program works for extracting texts from images. When I tried working with images with cursive handwriting, - it doesn't work. What are the procedures I have to do to add cursive extraction capabilities?

**Comment by robertknight** [comment, marked as answer]
You would need to train, or more likely fine-tune, the recognition model (and maybe the detection one) on a dataset of handwriting images. The [ocrs-models](https://github.com/robertknight/ocrs-models/blob/main/docs/training.md) repository has notes on how training of the current models was done. The documentation assumes some experience doing ML model training in PyTorch.

**Reply by kmgbarroga** [reply]
I see, this is informative thank you! 
Unfortunately I have zero experience when it comes to ML. It seems I need to grasp the fundamentals first.
By the way, I noticed that one of the requirements to do it is a graphics card? Any alternative that I can use? My pc is only a small dell optiplex 7050, i3-7100 and 16gb of ram.

**Reply by robertknight** [reply]
For ML training the alternative to owning a GPU is to rent one in the cloud, by the hour, from a service such as RunPod or AWS. Their cheapest options will be good enough. You _can_ run the training script on a CPU, and this is useful for testing that it works, but it will be very slow. What I usually do is to do a small test run on a sliver of the training data locally on the CPU to check the code runs, then do the actual training in the cloud on a GPU.

**Reply by kmgbarroga** [reply]
I see, thank you for the information! It's much appreciated.
